src/api/app/controllers/staging/staged_requests_controller.rb

```diff
@@ -62,21 +63,33 @@ def xml_hash
   end
 
   def set_staging_project
-    @staging_project = Project.get_by_name(params[:staging_project_name])
+    @staging_project = @staging_workflow.staging_projects.find_by(name: params[:staging_project_name])
+    return if @staging_project
+    render_error(
+      status: 404,
+      errorcode: 'not_found',
+      message: "Staging Project '#{params[:staging_project_name]}' doesn't exist or it doesn't belong to the Staging: '#{@staging_workflow.project}'"
```

Comment:
The second part of this message is not true anymore. 

> "Staging Project '#{params[:staging_project_name]}' doesn't exist or it doesn't belong to the Staging: '#{@staging_workflow.project}'"

"Staging Project '#{params[:staging_project_name]}' not found in in Staging: '#{@staging_workflow.project}'"